$OpenBSD: patch-src_lib_crypto_OSSLCryptoFactory_cpp,v 1.1.1.1 2018/11/14 22:23:07 pvk Exp $

ENGINE_load_rdrand() was removed from libcrypto in 2014:
https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/lib/libcrypto/engine/engine.h.diff?r1=1.25&r2=1.26&f=h

OpenDNSSEC GH commit:
https://github.com/opendnssec/SoftHSMv2/pull/423/commits/308b0b2760d6cb218003768747346d31764f1cfe

Index: src/lib/crypto/OSSLCryptoFactory.cpp
--- src/lib/crypto/OSSLCryptoFactory.cpp.orig
+++ src/lib/crypto/OSSLCryptoFactory.cpp
@@ -141,8 +141,10 @@ OSSLCryptoFactory::OSSLCryptoFactory()
 	// Initialise OpenSSL
 	OpenSSL_add_all_algorithms();
 
+#if !( OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER) )
 	// Make sure RDRAND is loaded first
 	ENGINE_load_rdrand();
+#endif
 	// Locate the engine
 	rdrand_engine = ENGINE_by_id("rdrand");
 	// Use RDRAND if available
